tests/PHPStan/Analyser/nsrt/get-defined-vars.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace GetDefinedVars;
+
+use function PHPStan\Testing\assertType;
+
+function doFoo(int $param) {
+	$local = "foo";
+	assertType('array{param: int, local: \'foo\'}', get_defined_vars());
+	assertType('array{\'param\', \'local\'}', array_keys(get_defined_vars()));
+	if(true) {
```

Comment:
use 

```suggestion
	if(rand(0,1) {
```

to have a condition which cannot be inferred at static analysis time

Reply:
I would keep both to test how it behaves when types can be inferred and when not.

Reply:
Test added and it revealed that current implementation did not handled maybe defined variables. So I fixed it. Thanks for pointing out that.